File: APP/lib/payment.dart
import 'package:app/user.dart';
import 'dart:convert';

class Payment {
  User fromUser;
  User toUser;
  double amount = 0.0;

  Payment(this.fromUser, this.toUser, this.amount);

  factory Payment.fromJson(dynamic json){
    User fromUser =  User.fromJson(json["from_user"]);
    User toUser =  User.fromJson(json["to_user"]);
    return Payment(fromUser, toUser, json["amount"] as double);
  }

  @override
  String toString() {
    String fromName = fromUser.name;
    String toName = toUser.name;
    return "$fromName -> $toName: \$$amount";
  }
}

File: API/domain/payment.py
class Payment:
    from_user = None
    to_user = None
    amount = 0.0

    def __init__(self, from_user: int, to_user: int, amount: float):
        self.from_user = from_user
        self.to_user = to_user
        self.amount = amount

    def to_json(self):
        return {"from_user_id": self.from_user, "to_user_id": self.to_user, "amount": self.amount}
